File: algorithms/algorithms-python/leetcode/Question_0001_Two_Sum.py
#!/usr/bin/python
# coding: utf-8

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        for i, v in enumerate(nums):
            diff = target - v
            if diff in m:
                return [m[diff], i]
            m[v] = i
        return None

File: algorithms/algorithms-cplusplus/leetcode/Question_0001_Two_Sum.cc
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
 public:
  vector<int> twoSum(vector<int>& nums, int target) {
    vector<int> res;
    unordered_map<int, int> m;
    for (int i = 0; i < nums.size(); i++) {
      int diff = target - nums[i];
      if (m.find(diff) != m.end()) {
        res.push_back(m[diff]);
        res.push_back(i);
        break;
      }
      m[nums[i]] = i;
    }
    return res;
  }
};
